### PR TITLE
Updating covidex to use pygaggle 0.0.3.1

### PR DESCRIPTION
**Pull request by ZanezZephyrs**
Updating the pygaggle dependency to a newer version, installing the most recent version from git because the most recent release in pip have some bugs that were already fixed in the git version of pygaggle.

With the update, some minor adjustments in the code were necessary.

### CODE REVIEW

File: api/app/services/ranker.py [hunk 1]
```diff
@@ -3,33 +3,27 @@
 import torch
 from pygaggle.model import CachedT5ModelLoader, T5BatchTokenizer
```
Comment (rodrigonogueira4) [inline api/app/services/ranker.py:4]: remove `CachedT5ModelLoader` since we don't need it anymore?

File: api/app/settings.py [hunk 2]
```diff
@@ -29,6 +29,7 @@ class Settings(BaseSettings):
     # T5 model settings
     neural_ranking: bool = True
     t5_model_dir: str = 'gs://neuralresearcher_data/covid/data/model_exp304'
```
Comment (rodrigonogueira4) [inline api/app/settings.py:31]: remove this since we don't need it anymore?

File: api/requirements.txt [hunk 3]
```diff
@@ -3,8 +3,8 @@ fastapi[all]==0.52.0
 freezegun==0.3.15
 hnswlib==0.3.4
 httpx==0.12.1
-pygaggle==0.0.1
+git+https://git@github.com/castorini/pygaggle.git@818a2487cb7af8ecdd63
```
Comment (edwinzhng) [inline api/requirements.txt:6]: Are we not able to specify 0.0.3.1?

**Review by edwinzhng — APPROVED** [review]
Thanks! Looks good to me other than the comments from @rodrigonogueira4

File: api/app/settings.py (hunk 2 above)
Comment (rodrigonogueira4) [reply api/app/settings.py:31]: This line also needs to be removed:
https://github.com/castorini/covidex/blob/c504a12c968952280f856d8ed32cac89e8b72797/api/.env.sample#L24

Comment (ZanezZephyrs) [reply api/app/settings.py]: I Removed then in the latest commit

File: api/app/services/ranker.py (hunk 1 above)
Comment (ZanezZephyrs) [reply api/app/services/ranker.py]: Done 👍

**Review by rodrigonogueira4 — APPROVED** [review]
LGTM, thanks for doing all this!

**Pull request merged**